### PR TITLE
add watch tag

### PR DESCRIPTION
**Pull request by omargho**
watching the specified Input/Source dirname for changes #833 
example usage : 
`apidoc -i example   -o ./doc --watch`

### CODE REVIEW

**Comment by omargho** [comment]
`nodemon` requires a nodeJS version >=8 in order to work as expected

File: bin/apidoc [hunk 1]
```diff
@@ -120,6 +123,19 @@ var options = {
     filterBy      : argv.filterBy,
 };
 
+if (argv.watch) {
+  nodemon({watch: options.src});
+  console.log('Watching mode is Activated\nListening for changes ...');
```
Comment (NicolasCARPi) [inline bin/apidoc]: there should not be a space before the ...

File: bin/apidoc [hunk 2]
```diff
@@ -120,6 +123,19 @@ var options = {
...
+  nodemon.on('quit', function () {
+    console.log('ApiDoc has quit');
```
Comment (NicolasCARPi) [inline bin/apidoc]: has exited sounds better

File: bin/apidoc [hunk 3]
```diff
@@ -120,6 +123,19 @@ var options = {
...
+    process.exit()
+  }).on('restart', function () {
+    console.log('Changes detected, Rebuilding the docs !!!');
```
Comment (NicolasCARPi) [inline bin/apidoc]: change  !!! in ...

File: bin/apidoc [hunk 4]
```diff
@@ -120,6 +123,19 @@ var options = {
...
+    apidoc.createDoc(options);
+    console.log('Listening for changes ...');
```
Comment (NicolasCARPi) [inline bin/apidoc]: no space before

**Comment by NicolasCARPi** [comment]
Seems to work as expected, thanks for your contribution! As for travis, should we just call the node v6 too old and remove it from the tests? I think it's reasonable to do so.

**Comment by omargho** [comment]
> As for travis, should we just call the node v6 too old and remove it from the tests? I think it's reasonable to do so.

i agree with you

**Pull request merged**